internal/reorder/deletebranch.go

```diff
@@ -0,0 +1,93 @@
+package reorder
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/spf13/pflag"
+
+	"github.com/aviator-co/av/internal/git"
+	"github.com/aviator-co/av/internal/utils/colors"
+	"github.com/aviator-co/av/internal/utils/errutils"
+)
+
+// DeleteBranchCmd is a command that deletes a branch.
+type DeleteBranchCmd struct {
+	Name string
+	// If true, delete the branch from Git as well as from the internal database.
+	// If false, only delete the branch metadata from the internal database.
+	DeleteRef bool
+}
+
+func (d DeleteBranchCmd) Execute(ctx *Context) error {
+	tx := ctx.DB.WriteTx()
+	tx.DeleteBranch(d.Name)
+	if err := tx.Commit(); err != nil {
+		return err
+	}
+
+	if !d.DeleteRef {
+		_, _ = fmt.Fprint(os.Stderr,
+			"Orphaned branch ", colors.UserInput(d.Name), ".\n",
+			"  - Run ", colors.CliCmd("git branch --delete ", d.Name),
+			" to delete the branch from your repository.\n",
+		)
+		return nil
+	}
+
+	_, err := ctx.Repo.Run(&git.RunOpts{
+		Args:      []string{"branch", "--delete", "--force", d.Name},
+		ExitError: true,
+	})
+	if exiterr, ok := errutils.As[*exec.ExitError](err); ok &&
+		strings.Contains(string(exiterr.Stderr), "not found") {
+		_, _ = fmt.Fprint(os.Stderr,
+			colors.Warning("Branch "), colors.UserInput(d.Name),
+			colors.Warning(" was already deleted.\n"),
+		)
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	_, _ = fmt.Fprint(os.Stderr,
+		"Deleted branch ", colors.UserInput(d.Name), ".\n",
+	)
+	return nil
+}
+
+func (d DeleteBranchCmd) String() string {
+	sb := strings.Builder{}
+	sb.WriteString("delete-branch ")
+	sb.WriteString(d.Name)
+	if d.DeleteRef {
```

Comment:
--delete-ref looks confusing? A branch is a ref.

Aside from that, do we want to have this command? It seems that simply deleting a line would be suffice to support deleting branches, and for I cannot think a good use case for this command.

Reply:
Yeah, I meant it to disambiguate deleting av's version of the branch and deleting the actual Git ref. It could also be `--delete-git` but I'm not sure if that's actually any clearer. Maybe `--delete-git-ref` would be the clearest of all.

But to your point, the only reason I implemented this as a command is convenience of implementation. It's nice to be able to just append the `DeleteBranch` commands to the reorder plan if we detect that the branches have been removed from the plan (which is what we do here). We probably don't even need to document that this exists. We only even need the flag version because the command is serialized as a string whenever we write the continuation to file.